commands/checkout.go

```diff
@@ -119,9 +119,19 @@ func transformCheckoutArgs(args *Args) error {
 				newBranchName = fmt.Sprintf("%s-%s", pullRequest.Head.Repo.Owner.Login, pullRequest.Head.Ref)
 			}
 		}
-		refSpec := fmt.Sprintf("refs/pull/%s/head:%s", id, newBranchName)
 		newArgs = append(newArgs, newBranchName)
-		args.Before("git", "fetch", baseRemote.Name, refSpec)
+
+		ref := fmt.Sprintf("refs/pull/%s/head", id)
+		args.Before("git", "fetch", baseRemote.Name, fmt.Sprintf("%s:%s", ref, newBranchName))
+
+		remote := baseRemote.Name
+		mergeRef := ref
+		if pullRequest.MaintainerCanModify && pullRequest.Head.Repo != nil {
+			remote = pullRequest.Head.Repo.SshUrl
+			mergeRef = fmt.Sprintf("refs/heads/%s", pullRequest.Head.Ref)
+		}
+		args.Before("git", "config", fmt.Sprintf("branch.%s.remote", newBranchName), remote)
+		args.Before("git", "config", fmt.Sprintf("branch.%s.merge", newBranchName), mergeRef)
```

Comment:
If the PR is coming from the same repository, `headRemote` won't be `nil` and we'll hit the previous branch. But if it's coming from a different repository but is *not* maintainer-modifiable, we still want to set the remote so `git pull` works.